contracts/LpSugar.vy

```diff
@@ -227,15 +297,35 @@ def forSwaps(_limit: uint256, _offset: uint256) -> DynArray[SwapLp, MAX_POOLS]:
         break
 
       pool_addr: address = factory.allPools(pindex)
-
-      pool: IPool = IPool(pool_addr)
-
-      if pool.reserve0() > 0:
+      type: int24 = -1
+      token0: address = empty(address)
+      token1: address = empty(address)
+      reserve0: uint256 = 0
+
+      # to-do: create preferred way of differentiating between v2 and v3 pools
+      is_cl_pool: bool = False
+
+      if is_cl_pool:
```

Comment:
I think we can use the factory `getPool()`, this could save us a call to `stable()` then:
https://github.com/velodrome-finance/contracts-private/blob/main/contracts/interfaces/factories/IPoolFactory.sol#L30-L34

We'll just call `token0()` and `token1()` prior to this.

If it doesn't return a blank address for the call with 0 or 1, it means it's a CL pool :thinking:

Reply:
The interface is the same, we've reserved the v2 factory exactly for the tick space/fee reason :)

Reply:
Not 100% sure, though it's unlikely that we'll ever pass a negative value there, so it should probably work as `uint24`.